src/main/java/gregtech/modules/ModuleManager.java

```diff
@@ -12,29 +12,36 @@
 import net.minecraftforge.fml.common.event.*;
 
 import com.google.common.base.Preconditions;
-import com.google.common.collect.ImmutableList;
+import it.unimi.dsi.fastutil.objects.Object2ReferenceLinkedOpenHashMap;
+import it.unimi.dsi.fastutil.objects.ObjectLinkedOpenHashSet;
+import it.unimi.dsi.fastutil.objects.ReferenceLinkedOpenHashSet;
+import it.unimi.dsi.fastutil.objects.ReferenceOpenHashSet;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.Unmodifiable;
 
 import java.io.File;
+import java.lang.reflect.InvocationTargetException;
 import java.util.*;
 import java.util.stream.Collectors;
 
-public class ModuleManager implements IModuleManager {
+public final class ModuleManager implements IModuleManager {
 
     private static final ModuleManager INSTANCE = new ModuleManager();
     private static final String MODULE_CFG_FILE_NAME = "modules.cfg";
     private static final String MODULE_CFG_CATEGORY_NAME = "modules";
     private static File configFolder;
 
-    private Map<String, IModuleContainer> containers = new LinkedHashMap<>();
-    private final Map<ResourceLocation, IGregTechModule> sortedModules = new LinkedHashMap<>();
-    private final Set<IGregTechModule> loadedModules = new LinkedHashSet<>();
+    private Map<String, IModuleContainer> containers = new Object2ReferenceLinkedOpenHashMap<>();
+    private final Map<ResourceLocation, IGregTechModule> sortedModules = new Object2ReferenceLinkedOpenHashMap<>();
+    private final Set<IGregTechModule> loadedModules = new ReferenceOpenHashSet<>();
```

Comment:
This makes the order of modules non-deterministic again, potentially causing modules to fire their events in an arbitrary order and cause packet id mismatches on server/client